### ISSUE TITLE
Observing difference in outputs from decoder with IO bindings.

### ISSUE THREAD

**Issue by VikasOjha666** [microsoft/onnxruntime#11096]
**Describe the bug**
Was trying to implement IO bindings for the decoder part of the model. Used the same code from FastT5  project repo on GitHub to convert the model to ONNX. After loading the model and making predictions using the decoder session directly the output appears to be fine but with inputs bound the result is coming to be different. I am trying to implement IO bindings for T5 with past_key_values as I observed that the model without past_key_values was becoming slow for longer sequences.

**Urgency**
It's quite urgent for our organization as we have a similar unoptimized torch model in production.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- ONNX Runtime installed from (source or binary): From PyPI.
- ONNX Runtime version: Tested with both version 1.10 and 1.11
- Python version: 3.7.12
- Visual Studio version (if applicable):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 11.2
- GPU model and memory: Tesla K80

Code:
def dec_pred_with_io_bindings(input_ids, encoder_attention_mask, encoder_hidden_states, past_key_values_dict,dec_session):
  dec_io_binding = dec_session.io_binding()
  dec_io_binding.bind_input(name="input_ids",
                          device_type="cuda",
                          device_id=0,
                          element_type=np.longlong,
                          shape=list(input_ids.shape),
                          buffer_ptr=input_ids.data_ptr())
  dec_io_binding.bind_input(name="encoder_attention_mask",
                          device_type="cuda",
                          device_id=0,
                          element_type=np.longlong,
                          shape=list(encoder_attention_mask.shape),
                          buffer_ptr=encoder_attention_mask.data_ptr())
                        
  dec_io_binding.bind_input(name="encoder_hidden_states",
                          device_type="cuda",
                          device_id=0,
                          element_type=np.float32,
                          shape=list(encoder_hidden_states.shape),
                          buffer_ptr=encoder_hidden_states.data_ptr())
  

  for key,val in past_key_values_dict.items():
    dec_io_binding.bind_input(name=str(key),
                                      device_type="cuda",
                                      device_id=0,
                                      element_type=np.float32,
                                      shape=list(val.shape),
                                      buffer_ptr=val.data_ptr())
  
  #Bind outputs.
  for arg in dec_session.get_outputs():
    dec_io_binding.bind_output(arg.name, "cuda")
    
  dec_session.run_with_iobinding(dec_io_binding)
  ort_output = dec_io_binding.get_outputs()

  logits=ort_output[0]

  list_pkv = tuple(torch.from_numpy(x.numpy()).cuda() for x in ort_output[1:])

  out_past_key_values = tuple(
      list_pkv[i : i + 4] for i in range(0, len(list_pkv), 4)
  )

  return torch.from_numpy(logits.numpy()).cuda(),out_past_key_values

The past_key_val_dict contains values in format like this {''pkv_0:pkv_val.......} which contains past_key_values of the model.

When running the model with the session run command the output is as expected but different for IO bindings.
@tianleiwu

**Comment by tianleiwu** [comment]
@[VikasOjha666](https://github.com/VikasOjha666), It is likely duplicated with https://github.com/microsoft/onnxruntime/issues/10992.  Input tensor need to be contiguous for IO binding.

You can follow our helper class to config io binding:
https://github.com/microsoft/onnxruntime/blob/8db180c245512f5bb5bdefc0f8b643ad6ff51294/onnxruntime/python/tools/transformers/io_binding_helper.py#L122

**Comment by VikasOjha666** [comment]
@tianleiwu Thanks once again for helping out to solve the problem like you always do it will help me a lot to take the model into production quickly. The problem is solved now. Making the tensors contiguous solved it.
